### PR TITLE
fix: replace PPID to using PID to get initial process

### PR DESCRIPTION
**Pull request by weizhe0422**
Source issue: [#2380](https://github.com/longhorn/longhorn/issues/2380)

Original is using PPID to determine initial process, since parent process of initial process will be 0. So it will make the log path always be `/tmp/ganesha.log`. 

Therefor, fix to modify as getting the PID to determine initial process.

### CODE REVIEW

**Comment by innobead** [comment]
@weizhe0422 Please add the source issue in the description as well.

File: pkg/server/nfs/nfs_server.go [hunk 1]
```diff
@@ -151,7 +151,8 @@ func getUpdatedGaneshConfig(config []byte) []byte {
 		logPath string
 	)
 
-	if os.Getppid() == 1 {
+	logrus.Infof("check the initial PID to determine log path: %d", os.Getpid())
```
Comment (innobead) [inline pkg/server/nfs/nfs_server.go:154]: This is quite straightforward. Suggest no need to have this log.

File: pkg/server/nfs/nfs_server.go [hunk 2]
```diff
@@ -164,5 +165,6 @@ func getUpdatedGaneshConfig(config []byte) []byte {
 	}
 
 	template.Must(template.New("Ganesha_Config").Parse(string(config))).Execute(&tmplBuf, tmplVals)
+	logrus.Infof("default nfs-ganesha config: %v", tmplBuf.String())
```
Comment (innobead) [inline pkg/server/nfs/nfs_server.go:168]: Don't suggest to disclose the internal config in the log.

**Review by innobead — CHANGES_REQUESTED** [review]
After resolving the comments, we can merge this.

Please ensure the PR fully tested.

**Comment by weizhe0422** [comment]
self-test steps:
1. Create an `rwx` volumes and attach it to one of the nodes.
2. Create a PVC/PV for this volumes from Step 1.
3. Deploy the POD and assign its PVC name from step 2.
4. Type `kubectl get pods -n <your_namespace>` and it will create share-manager POD whose name like `share-manager-<Longhorn_volume_name>`
5. Type `kubectl logs share-manager-<Longhorn_volume_name> -n <your_namespace>`
6. It will show off the nfs-ganesha log in the console, and it also appear in support bundle.

image

**Comment by innobead** [comment]
Thanks this is great.

BTW,did the log look verbose? cc @derekbit

**Comment by weizhe0422** [comment]
It seems like it will record detail steps such like setting log level for every components.

image

**Review by innobead — APPROVED** [review]

**Pull request merged**